python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -449,7 +449,7 @@ def test_filters_inclusive_set(tempdir, use_legacy_dataset):
     dataset = pq.ParquetDataset(
         base_path, filesystem=fs,
         filters=[('integer', 'in', [1]), ('string', 'in', ('a', 'b')),
-                 ('boolean', 'not in', {False})],
+                 ('boolean', 'not in', {'False'})],
```

Comment:
Hmm, interesting. Why does passing the right type work for integers but not booleans?

Reply:
Before it got implicitly cast to string and compared that way, now the check added disallows that (hence the behavior difference). I think it's because we don't infer boolean partition values.